doc/source/dev/migration_to_sparray.rst

```diff
@@ -0,0 +1,88 @@
+### Migration from spmatrix to sparray
+
+The changeover from sparse matrices to sparse arrays involves changes
```

Comment:
Let's start by saying that this mirrors the change from np.matrix to np.array, which hopefully we can find some useful pages on the web for. I wasn't able to find a canonical "migration from np.matrix to np.array" page though.

Reply:
I believe there never really was a migration from np.matrix to np.array.
Stefan and I were not able to find such previous documentation.

Still, I've added text to this point at the top.